pkg/network/mbus.go

```diff
@@ -7,25 +7,20 @@ import (
 	"net"
 
 	"github.com/pkg/errors"
-	substrate "github.com/threefoldtech/tfchain/clients/tfchain-client-go"
 	"github.com/threefoldtech/tfgrid-sdk-go/rmb-sdk-go"
 	"github.com/threefoldtech/zos/pkg"
 	"github.com/threefoldtech/zos/pkg/environment"
+	"github.com/threefoldtech/zos/pkg/stubs"
 )
 
 // Authorized middleware allows only admins to make these calls
-func Authorized(mgr substrate.Manager, farmID uint32) (rmb.Middleware, error) {
-	sub, err := mgr.Substrate()
-	if err != nil {
-		return nil, err
-	}
-	defer sub.Close()
-	farm, err := sub.GetFarm(farmID)
+func Authorized(apiGateway *stubs.APIGatewayStub, farmID uint32) (rmb.Middleware, error) {
```

Comment:
I am wondering if it might be better if we pass the zbus.Client instead of a specific stub since it's more generic and other sub modules can use it.

On the other hand this code should eventually be moved to the apiGatway itself soon in the future so we can leave it this way for now.